src/Microsoft.AspNet.Mvc.ModelBinding/BinderMetadata/BindingSource.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    /// <summary>
+    /// A metadata object representing a source of data for model binding.
+    /// </summary>
+    [DebuggerDisplay("Source: {DisplayName}")]
+    public class BindingSource : IEquatable<BindingSource>
+    {
+        /// <summary>
+        /// A <see cref="BindingSource"/> for the request body.
+        /// </summary>
+        public static readonly BindingSource Body = new BindingSource(
+            "Body",
+            Resources.BindingSource_Body,
+            isGreedy: true,
+            isFromRequest: true);
+
+        /// <summary>
+        /// A <see cref="ApiParameterSource"/> for a custom model binder (unknown data source).
+        /// </summary>
+        public static readonly BindingSource Custom = new BindingSource(
+            "Custom",
+            Resources.BindingSource_Custom,
+            isGreedy: true,
+            isFromRequest: true);
+
+        /// <summary>
+        /// A <see cref="ApiParameterSource"/> for the request form-data.
```

Comment:
Did a search for this to make sure there aren't any more. Sorry about missing that